airflow/traces/utils.py

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from airflow.utils.hashlib_wrapper import md5
+
+if TYPE_CHECKING:
+    from airflow.models import DagRun, TaskInstance
+    from airflow.models.taskinstancekey import TaskInstanceKey
+
+TRACE_ID = 0
+SPAN_ID = 16
+
+log = logging.getLogger(__name__)
+
+
+def _gen_id(seeds: list[str], as_int: bool = False, type: int = TRACE_ID) -> str | int:
+    seed_str = "_".join(seeds).encode("utf-8")
+    hash_hex = md5(seed_str).hexdigest()[type:]
+    return int(hash_hex, 16) if as_int else hash_hex
+
+
+def gen_trace_id(dag_run: DagRun, as_int: bool = False) -> str | int:
+    """Generate trace id from DagRun."""
+    return _gen_id([dag_run.dag_id, dag_run.run_id, str(dag_run.start_date.timestamp())], as_int)
+
+
```

Comment:
Now that you see it in action, how do you feel about this helper method way?   I generally go for positional arguments like you did here, but I wonder if using arg names in these might help clear it up a little?   I'll leave it up to you, but I wonder if it would be easeier tor ead as 

```
def gen_trace_id(dag_run: DagRun, as_int: bool = False) -> str | int:
    """Generate trace id from DagRun."""
    return _gen_id(
            seeds=[dag_run.dag_id, dag_run.run_id, str(dag_run.start_date.timestamp())], 
            as_int,
        )
```

The way it is, it's easy to lose the `as_int` in the visual clutter of the list, and adding the trailing comma in this case enforces the linter to separate them by line and adding "seeds=" maybe helps a bit visually here?  If you change "as_int" to an integer length as you mentioned, then using the name there would go a long way to making it visually distinct as well, I think.....   

I don't know, just a thought, feel free to ignore.  Style and visual tricks like that are pretty much up tot he individual.

Reply:
I think it's valid idea, and definitely looks much more readible.